server/notification-providers/discord.js

```diff
@@ -41,24 +56,7 @@ class Discord extends NotificationProvider {
                         title: "❌ Your service " + monitorJSON["name"] + " went down. ❌",
                         color: 16711680,
                         timestamp: heartbeatJSON["time"],
-                        fields: [
-                            {
-                                name: "Service Name",
-                                value: monitorJSON["name"],
-                            },
-                            {
-                                name: monitorJSON["type"] === "push" ? "Service Type" : "Service URL",
-                                value: this.extractAdress(monitorJSON),
```

Comment:
If you have the capacity for this:
Please add a button to the embed to visit the address and the monitor. That is much more usable.

Here is a PR that does what I think is best https://github.com/louislam/uptime-kuma/pull/4538

I think this should be possible going by https://discordjs.guide/message-components/buttons.html#building-buttons

Reply:
Components (buttons) on Discord are weird, you can only send them as a webhook if the webhook is created (owned) by a bot.

[API Ref](https://discord.com/developers/docs/resources/webhook#execute-webhook-jsonform-params) 

Notice the `*` next to components for "Requires an application-owned webhook."

Reply:
What would happen to non-application owned webhooks if we were to add buttons now and application owned webhooks later? Does discord reject that input or ignore it?

Application owned webhooks are another PR, I would be more than happy to walk you through our code if you can explain how they work. Docs on that seem quite sparse..
I don't know what having an app vs a webhook would entail.

Reply:
Discord would just ignore the components on non application webhooks, they accept the input so it's no problem there. That's why I offered to add the buttons anyway just with the warning that until automation or some kind of docs callout is added most people wouldn't really see it. 

Application owned webhooks are either webhooks created by a Bot account or webhooks created by a user going through the OAuth flow. Being self hosted software, the Bot account method is probably what we'd want to do - accepting a bot token and a channel ID similar to the Telegram notifier. 

Additionally we could use the Bot account to send messages directly without making a webhook as messages created by bots can use components (buttons) without an issue. Webhooks would give the configuration of the username and profile picture of the sent message to Uptime Kuma while bots give the configuration to the owner of the bot application.

Reply:
Then lets add the buttons now and add application owned webhooks/ bot messages later down the line 👍🏻